### PR TITLE
feat: improve decrease of validator set when changing the network par…

### LINKED ISSUE #7331: Controlled decrease of the number of validators upon network parameter change
# Feature Overview

When the network parameter controlling the number of validators in the network is decreased we want to reduce it one validator an epoch as long as the remaining number of validators is sufficiently large to be able to update the multisig contract. 

This means if the number of validators is going down from n to k:

1. sort the set of n validators by decreasing rank
2. remove the lowest ranking validator from consensus validators set
3. if this validator is not a signer on the multisig and more validators need to be removed to make n==k, keep removing validators.
4. if the validator is a signer and removing it will still leave enough validators such that `num_remaining * 1000/(num_signers) > threshold ` then remove the validator.
5. repeat in the next epoch until no more validators can be removed from the tendermint validator set or the number of validators in the set is equal to the value of the network parameter

## Specs
PR - https://github.com/vegaprotocol/specs/pull/1518/files

### PR DESCRIPTION
**Pull request by ze97286**
…ameter

Closes #7331

### CODE REVIEW

**Review by davidsiska-vega — DISMISSED** [review]
I think it matches what's proposed here. Anyway, let's get it in!

File: core/validators/validator_score_test.go
Comment (jeremyletang) [inline core/validators/validator_score_test.go:62]:
```suggestion
	return maps.Keys(t.validators)
```

**Review by jeremyletang — APPROVED** [review]
Approved, but you might want to wait for a review from @wwestgarth

**Review by wwestgarth — APPROVED** [review]
Code looks fine, but I am suprised there were no failures on the full run. Theres a test `test_pending_not_promoted` that drops the number of TM validators from 5 -> 3 without removing any signers and it is still passing.

You've added a unittest for this exact case though and it works as expected (not allowing the full drop until signers are removed), so I'm just going to assume its a system-test issue for now. I'll make a note about it when the QA ticket for the ACs pops up.

**Comment by ze97286** [comment]
> Code looks fine, but I am suprised there were no failures on the full run. Theres a test `test_pending_not_promoted` that drops the number of TM validators from 5 -> 3 without removing any signers and it is still passing.
> 
> You've added a unittest for this exact case though and it works as expected (not allowing the full drop until signers are removed), so I'm just going to assume its a system-test issue for now. I'll make a note about it when the QA ticket for the ACs pops up.

let me rerun this system test because the full run precedes the latest code which includes this test you're referring to

**Pull request merged**
